packages/block-library/src/navigation/edit/index.js

```diff
@@ -293,15 +314,17 @@ function Navigation( {
 		setIsPlaceholderShown( ! isEntityAvailable );
 	}, [ isEntityAvailable ] );
 
-	// If the ref no longer exists the reset the inner blocks
-	// to provide a clean slate.
+	// If the ref no longer exists the reset the inner blocks to provide a
+	// clean slate.
 	useEffect( () => {
-		if ( ref === undefined && innerBlocks.length > 0 ) {
+		if (
+			! hasUncontrolledInnerBlocks &&
+			controlledInnerBlocks?.length > 0 &&
```

Comment:
These new conditionals cause a regression when you're trying to start a new menu from an existing Menu that previously had uncontrolled inner blocks and has been converted to controlled blocks.

Again this is because the block editor store isn't in sync and shows outdated inner blocks.

Reply:
This will start working _if_ we implement a change which gets the block editor store in sync with the controlled block (i.e. inner blocks should be removed once we've converted to an entity and controlled inner blocks).

Reply:
Yeah, that makes sense, thanks for catching it I'll look into a fix 👍